ui/apps/everest/.e2e/release/demand-backup-psmdb-sharding.e2e.ts

```diff
@@ -0,0 +1,344 @@
+// limitations under the License.
+
+import { expect, test } from '@playwright/test';
+import {
+  deleteDbCluster,
+  gotoDbClusterBackups,
+  gotoDbClusterRestores,
+} from '@e2e/utils/db-clusters-list';
+import { getTokenFromLocalStorage } from '@e2e/utils/localStorage';
+import { getClusterDetailedInfo } from '@e2e/utils/storage-class';
+import {
+  moveForward,
+  submitWizard,
+  populateBasicInformation,
+  populateResources,
+  populateAdvancedConfig,
+  populateMonitoringModalForm,
+} from '@e2e/utils/db-wizard';
+import { EVEREST_CI_NAMESPACES } from '@e2e/constants';
+import {
+  waitForStatus,
+  waitForDelete,
+  findRowAndClickActions,
+} from '@e2e/utils/table';
+import { checkError } from '@e2e/utils/generic';
+import {
+  deleteMonitoringInstance,
+  listMonitoringInstances,
+} from '@e2e/utils/monitoring-instance';
+import { clickOnDemandBackup } from '@e2e/pr/db-cluster-details/utils';
+import {
+  queryPSMDB,
+  prepareMongoDBTestDB,
+  validateMongoDBSharding,
+  configureMongoDBSharding,
+  dropTestDB,
+  queryTestDB,
+} from '@e2e/utils/db-cmd-line';
+
+export let isShardingEnabled = false;
+
+const {
+  MONITORING_URL,
+  MONITORING_USER,
+  MONITORING_PASSWORD,
+  SELECT_DB,
+  SELECT_SIZE,
+} = process.env;
+let token: string;
+
+const db = 'psmdb';
+const size = 3;
+
+test.describe.configure({ retries: 0 });
+
+test.describe(
+  'Demand backup psmdb',
+  {
+    tag: '@release',
+  },
+  () => {
+    test.skip(
+      () =>
+        (SELECT_DB !== db && !!SELECT_DB) ||
+        (SELECT_SIZE !== size.toString() && !!SELECT_SIZE)
+    );
+    test.describe.configure({ timeout: 720000 });
+
+    const clusterName = `${db}-${size}-dembkp`;
+
+    let storageClasses = [];
+    const namespace = EVEREST_CI_NAMESPACES.EVEREST_UI;
+    const monitoringName = `${db}-${size}-pmm`;
+    const baseBackupName = `dembkp-${db}-${size}`;
+
+    test.beforeAll(async ({ request }) => {
+      token = await getTokenFromLocalStorage();
+
+      const { storageClassNames = [] } = await getClusterDetailedInfo(
+        token,
+        request
+      );
+      storageClasses = storageClassNames;
+    });
+
+    test.afterAll(async ({ request }) => {
+      // we try to delete all monitoring instances because cluster creation expects that none exist
+      // (monitoring instance is added in the form where the warning that none exist is visible)
+      const monitoringInstances = await listMonitoringInstances(
+        request,
+        namespace,
+        token
+      );
+      for (const instance of monitoringInstances) {
+        await deleteMonitoringInstance(
+          request,
+          namespace,
+          instance.name,
+          token
+        );
+      }
+    });
+
+    test(`Cluster creation [${db} size ${size}]`, async ({ page, request }) => {
+      expect(storageClasses.length).toBeGreaterThan(0);
+
+      await page.goto('/databases');
+      await page.getByTestId('add-db-cluster-button').waitFor();
```

Comment:
i think this part should be moved to utils to avoid duplication  🤔 we call the cluster creation button in a lot of places